packages/core/src/main/broadcast/BroadcastService.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {APIClient} from '@wireapp/api-client';
+import {NewOTRMessage, OTRRecipients} from '@wireapp/api-client/dist/commonjs/conversation/';
+import {UserPreKeyBundleMap} from '@wireapp/api-client/dist/commonjs/user/';
+import {GenericMessage} from '@wireapp/protocol-messaging';
+import {ConversationService} from '../conversation/root';
+import {CryptographyService} from '../cryptography/root';
+
+class BroadcastService {
+  constructor(
+    private readonly apiClient: APIClient,
+    private readonly conversationService: ConversationService,
+    private readonly cryptographyService: CryptographyService
+  ) {}
+
+  private async getPreKeyBundle(teamId: string, skipOwnClients = false): Promise<UserPreKeyBundleMap> {
+    const {members: teamMembers} = await this.apiClient.teams.member.api.getMembers(teamId);
+
+    let members = teamMembers.map(member => ({id: member.user}));
+
+    if (skipOwnClients) {
+      const selfUser = await this.apiClient.self.api.getSelf();
+      members = members.filter(member => member.id !== selfUser.id);
+    }
+
+    const preKeys = await Promise.all(members.map(member => this.apiClient.user.api.getUserPreKeys(member.id)));
+
+    return preKeys.reduce((bundleMap: UserPreKeyBundleMap, bundle) => {
+      bundleMap[bundle.user] = {};
+      for (const client of bundle.clients) {
+        bundleMap[bundle.user][client.client] = client.prekey;
+      }
+      return bundleMap;
+    }, {});
+  }
+
+  public async broadcastGenericMessage(teamId: string, genericMessage: GenericMessage): Promise<void> {
+    const clientId = this.conversationService.getClientID();
+
+    const plainTextArray = GenericMessage.encode(genericMessage).finish();
+    const preKeyBundle = await this.getPreKeyBundle(teamId);
+    const recipients = await this.cryptographyService.encrypt(plainTextArray, preKeyBundle);
+
+    return this.sendOTRBroadcastMessage(clientId, recipients, plainTextArray);
+  }
+
+  private async sendOTRBroadcastMessage(
+    sendingClientId: string,
+    recipients: OTRRecipients,
+    plainTextArray: Uint8Array,
+    data?: any
```

Comment:
Can this be specified?

Reply:
Unfortunately not, we also have this in [ConversationService.ts#L479](https://github.com/wireapp/wire-web-packages/blob/master/packages/core/src/main/conversation/ConversationService.ts#L479).